app/src/main/kotlin/com/mapzen/erasermap/view/RouteModeView.kt

```diff
@@ -431,6 +433,12 @@ class RouteModeView : LinearLayout, RouteViewController, ViewPager.OnPageChangeL
         initInstructionAdapter()
         this.visibility = View.VISIBLE
         routePresenter.onRouteResume(route)
+        Handler().postDelayed( {
+            if (routePresenter.currentSnapLocation != null) {
+                showRouteIcon(routePresenter.currentSnapLocation!!)
```

Comment:
Introducing a read-only local variable might be a better way to express this without resorting to forced de-reference using `!!` operator:

```kotlin
        Handler().postDelayed( {
            val currentSnapLocation = routePresenter.currentSnapLocation
            if (currentSnapLocation != null) {
                showRouteIcon(currentSnapLocation)
            }
            centerMapOnCurrentLocation()
        }, 100)
```